src/circuit.rs

```diff
@@ -74,10 +138,12 @@ where
             prover
                 .prover_key
                 .expect("Unexpected error. Missing ProverKey in compilation"),
-            verifier
-                .verifier_key
-                .expect("Unexpected error. Missing VerifierKey in compilation"),
-            pi_pos,
+            VerifierData::new(
```

Comment:
True. Let me change that.

Reply:
Sorry, The expects are correct. At least for me this should be a panic. 
Here we're not unwrapping a result but  an `Option`. And it should contain the T's. Therefore I'd leave it as is. SInce the expected behaviour if this happens should be a panic since it's unexpected behaviour

Reply:
We should always panic as little as possible. In this case, its trivial to not panic

Reply:
Again, this is not an Error. This is unexpected behaviour. This should not happen. If after a `preprocess()` call the keys don't reside inside of the `Option` something really bad happened. 
Idk if this leaves the program on a state where you can simply continue operating.